stats/stats-server/src/server.rs

```diff
@@ -56,6 +59,106 @@ fn grpc_router<S: StatsService>(
         .add_service(StatsServiceServer::from_arc(stats))
 }
 
+fn is_threshold_passed(
+    threshold: &ToggleableThreshold,
+    float_value: Option<String>,
+    value_name: &str,
+) -> Result<bool, anyhow::Error> {
+    let threshold = if threshold.enabled {
+        threshold.threshold
+    } else {
+        return Ok(true);
+    };
+    let value = float_value
+        .map(|s| s.parse::<f64>())
+        .transpose()
+        .context(format!("Parsing `{value_name}`"))?;
+    let Some(value) = value else {
+        anyhow::bail!("Received `null` value of `{value_name}`. Can't determine indexing status.",);
+    };
+    if value < threshold {
+        info!(
+            threshold = threshold,
+            current_value = value,
+            "Threshold for `{value_name}` is not satisfied"
+        );
+        Ok(false)
+    } else {
+        info!(
+            threshold = threshold,
+            current_value = value,
+            "Threshold for `{value_name}` is satisfied"
+        );
+        Ok(true)
+    }
+}
+
+fn is_retryable_code(status_code: &reqwest::StatusCode) -> bool {
+    matches!(
+        *status_code,
+        StatusCode::INTERNAL_SERVER_ERROR
+            | StatusCode::SERVICE_UNAVAILABLE
+            | StatusCode::GATEWAY_TIMEOUT
+            | StatusCode::TOO_MANY_REQUESTS
+            | StatusCode::IM_A_TEAPOT
+    )
+}
+
+async fn wait_for_blockscout_indexing(
+    api_config: blockscout_client::Configuration,
+    wait_config: StartConditionSettings,
+) -> Result<(), anyhow::Error> {
+    loop {
+        match blockscout_client::apis::main_page_api::get_indexing_status(&api_config).await {
+            Ok(result)
+                if is_threshold_passed(
+                    &wait_config.blocks_ratio,
+                    result.indexed_blocks_ratio.clone(),
+                    "indexed_blocks_ratio",
+                )? && is_threshold_passed(
+                    &wait_config.internal_transactions_ratio,
+                    result.indexed_internal_transactions_ratio.clone(),
+                    "indexed_internal_transactions_ratio",
+                )? =>
+            {
+                info!("Blockscout indexing threshold passed");
+                return Ok(());
+            }
+            Ok(_) => {}
+            Err(blockscout_client::Error::ResponseError(r)) if is_retryable_code(&r.status) => {
+                warn!("Error from indexing status endpoint: {r:?}");
+            }
+            Err(e) => {
+                return Err(e).context("Requesting indexing status");
+            }
+        }
+        info!(
+            "Blockscout is not indexed enough. Checking again in {} secs",
+            wait_config.check_period_secs
+        );
+        sleep(Duration::from_secs(wait_config.check_period_secs.into())).await;
+    }
+}
+
+async fn init_api(settings: &Settings) -> anyhow::Result<Option<blockscout_client::Configuration>> {
```

Comment:
init_api sounds like we are initializing stats api for some reason. lets name it init_blockscout_client